File: app/src/main/java/com/derlados/computerconf/Fragments/MainMenuFragment.java
package com.derlados.computerconf.Fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.derlados.computerconf.Fragments.PageFragment.MenuPageAdapter;
import com.derlados.computerconf.Fragments.PageFragment.PageFragment;
import com.derlados.computerconf.R;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainMenuFragment extends Fragment implements BottomNavigationView.OnNavigationItemSelectedListener {

    private ViewPager pager;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_main_menu, container, false);
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);

        // По скольку дочерние фрагменты не реагируют на изменения видимости родителя - необходимо чтобы родителя сам их оповещал
        for (int i = 0; i < pager.getAdapter().getCount(); ++i) {
            Fragment fragment = ((MenuPageAdapter) pager.getAdapter()).getPageFragment(i);
            // Необходимо проверять создался ли объект, ибо создаются они лишь заранее на 1 влево и вправо в viewPager
            if (fragment != null)
                fragment.onHiddenChanged(hidden);
        }
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        // Настройка ViewPager-а для просмотра
        pager = this.getView().findViewById(R.id.fragment_main_menu_pager);
        pager.setAdapter(new MenuPageAdapter(getFragmentManager()));
        pager.setOffscreenPageLimit(pager.getAdapter().getCount());
        ((BottomNavigationView)getView().findViewById(R.id.fragment_main_menu_bottom_navigator)).setOnNavigationItemSelectedListener(this);
    }

    // Переход между страницами при помощи меню в нижней части экрана
    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.main_menu_bottom_navigator_action_builds:
                pager.setCurrentItem(PageFragment.PageMenu.BUILDS.ordinal(), false);
                break;
            case R.id.main_menu_bottom_navigator_action_shop:
                pager.setCurrentItem(PageFragment.PageMenu.SHOP.ordinal(), false);
                break;
            case R.id.main_menu_bottom_navigator_action_info:
                pager.setCurrentItem(PageFragment.PageMenu.INFO.ordinal(), false);
                break;
        }
        return false;
    }
}

File: app/src/main/java/com/derlados/computer_conf/VIews/MainMenuFragment.kt
package com.derlados.computerconf.VIews

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.derlados.computer_conf.R
import com.derlados.computerconf.VIews.PageFragment.MenuPageAdapter
import com.derlados.computerconf.VIews.PageFragment.PageFragment.PageMenu
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.fragment_main_menu.view.*

class MainMenuFragment : Fragment(), BottomNavigationView.OnNavigationItemSelectedListener {
    private var pager: ViewPager? = null
    private lateinit var fragment: View

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        fragment = inflater.inflate(R.layout.fragment_main_menu, container, false)

        return fragment
    }

    //TODO (По хорошему нужно делать через контроллер, однако функционал полностью под ЖЦ UI)
    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)

        // По скольку дочерние фрагменты не реагируют на изменения видимости родителя - необходимо чтобы родителя сам их оповещал
        for (i in 0 until (pager?.adapter?.count ?: 0)) {
            val fragment = (pager?.adapter as MenuPageAdapter?)?.getPageFragment(i)
            // Необходимо проверять создался ли объект, ибо создаются они лишь заранее на 1 влево и вправо в viewPager
            fragment?.onHiddenChanged(hidden)
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        // Настройка ViewPager-а для просмотра
        pager = fragment.fragment_main_menu_pager
        pager?.adapter = MenuPageAdapter(fragmentManager)
        pager?.offscreenPageLimit = pager?.adapter!!.count
        (fragment.fragment_main_menu_bottom_navigator as BottomNavigationView).setOnNavigationItemSelectedListener(this)
    }

    // Переход между страницами при помощи меню в нижней части экрана
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.main_menu_bottom_navigator_action_builds -> pager!!.setCurrentItem(PageMenu.BUILDS.ordinal, false)
            R.id.main_menu_bottom_navigator_action_shop -> pager!!.setCurrentItem(PageMenu.SHOP.ordinal, false)
            R.id.main_menu_bottom_navigator_action_info -> pager!!.setCurrentItem(PageMenu.INFO.ordinal, false)
        }
        return false
    }
}